app/scripts/controllers/permissions/index.js

```diff
@@ -510,6 +507,100 @@ export class PermissionsController {
     })
   }
 
+  /**
+   * Removes all known domains and their related permissions.
+   */
+  clearPermissions () {
+    this.permissions.clearDomains()
+    this.notifyAllDomains({
+      method: NOTIFICATION_NAMES.accountsChanged,
+      result: [],
+    })
+  }
+
+  /**
+   * Stores domain metadata for the given origin (domain).
+   * Deletes metadata for domains without permissions in a FIFO manner.
+   * Metadata is only deleted for domains with permissions on boot in order to
```

Comment:
Is this supposed to be _without_?

```suggestion
   * Metadata is only deleted for domains without permissions on boot in order to
```